src/main/java/pl/javastart/task/BubbleSort.java

```diff
@@ -4,6 +4,27 @@ public class BubbleSort {
 
     // ta metoda powinna sortować inputArray oraz zwracać informację o liczbie porównań, które były do tego potrzebne
     public int sort(int[] inputArray) {
-        return 0;
+        int count = 0;
+        int breakeCount = 0;
+        int lowerLoop = inputArray.length - 1;
+        for (int i = 0; i < inputArray.length; i++) {
+            for (int j = 0; j < lowerLoop; j++) {
+                if (inputArray[j] < inputArray[(j + 1 + inputArray.length) % inputArray.length]) {
+                    breakeCount++;
+                } else if (inputArray[j] > inputArray[j + 1]) {
+                    int tmp = inputArray[j + 1];
+                    inputArray[j + 1] = inputArray[j];
+                    inputArray[j] = tmp;
+                }
+                count++;
+            }
+            if (i == 0 && breakeCount == inputArray.length - 1) {
+                break;
+            } else if (i == 1 && breakeCount == inputArray.length * 2 - 4) {
+                break;
+            }
+            lowerLoop--;
```

Comment:
Wyniki są poprawne i za to plus, ale ciężko zrozumieć to przerywanie pętli. Można to znacząco uprościć wprowadzając dodatkową zmienną boolean. W pseudokodzie:
```
boolean sorted;
for(...) {
  sorted = true;
  for(...) {
    if (inputArray[j] > inputArray[j + 1]) {
        ...
        sorted = false;
    }
  if (sorted) {
     return ...
 ...
```
czyli "jeżeli w i-tej iteracji pętli zewnętrznej nie podmieniliśmy żadnych elementów, to możemy przerwać pętlę i zwrócić wynik, bo tablica jest posortowana.